core/src/main/scala/com/spotify/featran/transformers/MDL.scala

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2017 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.featran.transformers
+
+import com.spotify.featran.FeatureBuilder
+import com.spotify.featran.transformers.mdl.MDLPDiscretizer
+import com.spotify.featran.transformers.mdl.MDLPDiscretizer._
+import com.twitter.algebird._
+import spire.ClassTag
+
+import scala.collection.JavaConverters._
+import scala.util.Random
+
+case class MDLRecord[T](label: T, value: Double)
+import java.util.{TreeMap => JTreeMap}
+
+/**
+ * This Transformer computes the optimal number of buckets using
+ * minimum description length. That is an entropy measurement between
+ * the values and the tagets.
+ *
+ * The Transformer expects an MDLRecord where the first field is a label
+ * and the second value is the scalar that will be transformed into buckets.
+ *
+ * MDL is an iterative algorithm so all of the data needed to compute the buckets
+ * will need to be pulled into memory.  If you run into memory issues the sampleRate
+ * parameter will need to be lowered.
```

Comment:
do you have external pointers about this version MDL? would love to read more about it

Reply:
The core part of the code comes from this repo.  https://github.com/sramirez/spark-MDLP-discretization.  

Old school paper: http://sci2s.ugr.es/keel/pdf/algorithm/congreso/fayyad1993.pdf

Reply:
nice! will take a look - if anyone else is interested, there is a cleaner scan of the same paper here: http://web.donga.ac.kr/kjunwoo/files/Multi%20interval%20discretization%20of%20continuous%20valued%20attributes%20for%20classification%20learning.pdf